hydromt_wflow/utils.py

```diff
@@ -475,6 +599,16 @@ def _update_output_netcdf_grid(wflow_var, var_name):
         else:
             _warn_str(var_name, "netcdf_grid")
 
+    # Solve the var name
+    def _solve_var_name(var: str | dict, path: str, add: list):
+        if not isinstance(var, dict):
+            sep = "." if path else ""
+            add_str = ".".join(add) if add else ""
+            yield (var, path + sep + add_str)
+            return
+        for key, item in var.items():
+            yield from _solve_var_name(item, path, add + [key])
+
```

Comment:
It's not quite clear to me what this does or what it is used for. Could you add some comments? Also did you intend to make this a closure?